client/src/signup/SignUp.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import axios from 'axios';
+
+class SignUp extends React.Component {
+  state = {
+    username: '',
+    password: '',
+    department: ''
+  }
+
+  handleInputChange = e => {
+    this.setState({[e.target.name]: e.target.value});
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const endpoint = 'http://localhost:9090/api/auth/register';
+
+    axios.post(endpoint, this.state)
+         .then(res => {
+           console.log(res);
+         })
+         .catch(err => {
+           console.log(err);
+         });
+
+    this.props.history.push('/signin');
+  }
+
+  render() {
+    return (
+      <>
+        <h2>Sign Up</h2>
+        <form onSubmit={this.handleSubmit}>
+          <input
+            name = 'department'
+            id = 'department'
+            placeholder = 'Department'
+            value = {this.state.department}
+            onChange = {this.handleInputChange}
+            type = 'text'
+          />
+          <input
+            name = 'username'
+            id = 'username'
+            placeholder = 'Username'
+            value = {this.state.username}
+            onChange = {this.handleInputChange}
+            type = 'text'
+          />
+          <input
+            name = 'password'
+            id = 'password'
+            placeholder = 'Password'
+            value = {this.state.password}
+            onChange = {this.handleInputChange}
+            type = 'password'
+          />
```

Comment:
These inputs seem to be pretty repetitive; do you think you could make a component for them? For example's sake, here's a method to handle this:
```javascript
const makeInput = (name, type = 'text') => {
  <input 
    name={name}
    id={name}
    placeholder={name[0].toUpperCase() + name.slice(1)}
    value={this.state[name]}
    onChange={this.handleInputChange}
    type={type}
  />
};
```
and its usage:
```javascript
<form  onSubmit ={this.handleSubmit}>
  {this.makeInput('department')}
  {this.makeInput('username')}
  {this.makeInput('password', 'password')}
  <button type='submit'>SignUp</button>
</form>
```